src/Feature/FeatureTrace.hs

```diff
@@ -47,6 +47,10 @@ simplifyFeatureTraceOfNodes f t d = \v -> if member v d then simplifyFeatureTrac
 Calculates the presence condition of a node (third argument) in the given tree (first argument) with the given feature traces (second argument).
 See Equation 1 in the paper.
 -}
+-- TODO(benjamin) In the paper `trace node` is not included if `node` is
+-- `Mandatory`. Does this code need the following invariant?
+-- `all (== Nothing) [trace n | n <- nodes root, optionaltype n == Mandatory]`
```

Comment:
Yes. Actually, this isn't the only place that requires this invariant. This invariant should (almost) always hold for all `FeatureTrace`s with all `AST`s